### PR TITLE
Add deposit feature

### PR DESCRIPTION
**Pull request by xiaoge26**
Allow user to add deposit.

### CODE REVIEW

File: src/main/java/seedu/bankwithus/Account.java [hunk 1]
```diff
@@ -4,7 +4,7 @@ public class Account {
     protected String name;
     protected float balance;
 
-
+    private Ui ui;
```
Comment (tyuyang) [inline src/main/java/seedu/bankwithus/Account.java:7]: If you need the ui object just instantiate a new one in the function.

File: src/main/java/seedu/bankwithus/AccountList.java [hunk 2]
```diff
@@ -46,5 +46,16 @@ public String getAllAccountDetails() {
         return temp;
     }
 
+    public Account getCurrentAccount() {
+        return accounts.get(0);
+    }
 
+    public void depositMoney(String depositAmountString) throws NumberFormatException, NullPointerException {
```
Comment (tyuyang) [inline src/main/java/seedu/bankwithus/AccountList.java:53]: Funcyion never throws any exception since it is all in  the try catch block

Comment (tyuyang) [reply src/main/java/seedu/bankwithus/AccountList.java:53]: I suggest to let it just throw the exception, and let the parser handle the exception instead i.e. remove the try catch block here and put it in the parser class. I think you can also remove the ui object if you do this

File: src/main/java/seedu/bankwithus/BankWithUs.java [hunk 3]
```diff
@@ -94,6 +94,15 @@ public void run() throws IOException {
...
     }
 
+
+    public AccountList getAccountList() {
+        return accountList;
+    }
+
+    public void setAccountList(AccountList accountList) {
+        this.accountList = accountList;
+    }
```
Comment (tyuyang) [inline src/main/java/seedu/bankwithus/BankWithUs.java:104]: Not necessary in our program right now. Add this only if we really need this.

File: src/main/java/seedu/bankwithus/Parser.java [hunk 4]
```diff
@@ -16,24 +17,23 @@ public Parser(BankWithUs bwu) {
...
-                    bwu.isExitEntered = true;
-                    break;
-                default:
-                    throw new CommandNotFoundException();
-            }
-        }
+            break;
+        case "deposit":
+            bwu.getAccountList().depositMoney(args);
+            ui.showDepositMessage();
```
Comment (tyuyang) [inline src/main/java/seedu/bankwithus/Parser.java:32]: See the 2nd comment

**Review by tyuyang — COMMENTED** [review]
Actually I didn't find too many issues with this code. Do try to extract out the try catch block into the parser though. Currently if you enter in an invalid number, both the invalidNumberError and showDepositMessage will be called. Also make sure that the program works as expected if either invalid number or valid number is entered. Otherwise lgtm, ping once you are done doing the changes

**Comment by xiaoge26** [comment]
I'll do the improvement on another branch in my repo and close this pull request.

**Pull request closed without merging**